src/buffer/out/Row.cpp

```diff
@@ -433,7 +433,9 @@ OutputCellIterator ROW::WriteCells(OutputCellIterator it, const til::CoordType c
     auto colorStarts = gsl::narrow_cast<uint16_t>(columnBegin);
     auto currentIndex = colorStarts;
 
-    while (it && currentIndex <= finalColumnInRow)
+    // finalColumnInRow is 32-bit and currentIndex is 16-bit. If finalColumnInRow is higher
+    // than the 16-bit integer limit, we will loop indefinitely.
+    while (it && currentIndex <= saturated_cast<uint16_t>(finalColumnInRow))

```

Comment:
I have not tested this, but looking at just the code, it seems to me that simply removing the `narrow_cast` above would have the same effect of fixing this bug.

Reply:
Nice catch. Seems like `colorStarts` is used below for `_attr.replace()` and it requires a `size_type` which is 32-bit, so seems that it'll work.